tests/integration/conftest.py

```diff
@@ -809,4 +809,8 @@ async def traefik_application_fixture(model: Model, external_hostname: str):
         raise_on_error=False,
     )
 
-    yield traefik
+    status = await model.get_status(filters=[traefik.name])
+    unit = next(iter(status.applications[traefik.name].units))
+    traefik_address = status["applications"][traefik.name]["units"][unit]["address"]
```

Comment:
I think I would have preferred to move this into a helper function, as this may not be necessary for all upcoming tests and we would not need to return a tuple here.